shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -273,17 +273,21 @@ public void updateSemantics(ByteBuffer buffer, String[] strings) {
           new AccessibilityManager.AccessibilityStateChangeListener() {
             @Override
             public void onAccessibilityStateChanged(boolean accessibilityEnabled) {
-              if (accessibilityEnabled) {
-                accessibilityChannel.setAccessibilityMessageHandler(accessibilityMessageHandler);
-                accessibilityChannel.onAndroidAccessibilityEnabled();
+              if (accessibilityChannel.flutterJNI.isAttached()) {
```

Comment:
Are you saying that when you traced this on a Huawei phone, this onAccessibilityStateChanged callback happens sometimes after FlutterEngine.destroy()? Who called FlutterEngine.destroy() in this case? 

Seems to me like the easier fix is to make sure FlutterView.detachFromFlutterEngine which calls AccessibilityBridge.release which removes this accessibilityStateChangeListener before the FlutterEngine.destroy calls FlutterJNI.detachFromNativeAndReleaseResources no? 

Can you share more about what's happening between the FlutterEngine and your FlutterView in this case?

Reply:
Oh wait, @jason-simmons already made more or less the same comment in https://github.com/flutter/engine/pull/17292 lol :)

Reply:
> yes,i can recurrent this bug locally with Huawei ,and i use v1.7.8 io.flutter.view.FlutterView,
For safety reasons,Huawei will close Accessibility when app quit completely.If phone low memory and try to kill some background app,onAccessibilityStateChanged and FlutterEngine.destroy will occur at the same time ,this will greatly increase the probability of crash

Understood. Huawei closing accessibility and calling onAccessibilityStateChanged isn't an issue. I'm mostly curious about what triggered FlutterEngine.destroy but didn't trigger a FlutterActivity.onDestroy or FlutterFragment.onDetroyView (both of which triggers FlutterView.detachFromFlutterEngine)

Reply:
i just use io.flutter.view.FlutterView,not use FlutterActivity and FlutterFragment,Normally  it will removeAccessibilityStateChangeListener first,and then FlutterJNI.nativePlatformViewId=null.AccessibilityManager.notifyAccessibilityStateChanged() use post to call listener.onAccessibilityStateChanged(),so when onAccessibilityStateChanged and FlutterEngine.destroy  occur at the same time,it may be post->removeAccessibilityStateChangeListener->FlutterJNI.nativePlatformViewId=null->runnable.run(listener.onAccessibilityStateChanged),this will cause crash.